lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
+                                               reason=f"Requested role creation by {itx.user}")
+        except discord.Forbidden:
+            await itx.response.send_message("Unable to create a role because I am missing the Manage Roles permission!",
+                                            ephemeral=True)
+            return
+
+        await self.insert_role(itx, role)
+        view = ConfirmationView(message="", author_id=itx.user.id, delete_message_after=True)
+        await itx.response.edit_message(content="Created a new role. In order for this role to work "
+                                        "correctly, you must set permission overrides for every channel. "
+                                        "Would you like me to do this automatically?", view=view)
+        await view.wait()
+        if view.value is False:
+            # a message here
+            return
+
+        s, f, sk = await self.create_permission_overwrites(itx.guild, role)
+        if f >= 1:
+            content = f"Set {s+sk} permissions overwrites for this role. {f} channels failed to set permission "\
+                      "overrides!"
+        else:
+            content = f"Set {s+sk} permission overwrites for this role."
+
+        await itx.followup.send(content=content, ephemeral=True)
+        self.stop(interaction=itx)
+
+
+class GatekeeperChannelView(BasicMenuLikeView):
+    def __init__(self, role: discord.Role, channel: Optional[discord.TextChannel] = None, *,
+                 author_id: int, clear_view_after=False, delete_message_after=False,
+                 disable_components_after=True, timeout: float | None = 180):
+        super().__init__(author_id=author_id,
+                         clear_view_after=clear_view_after,
+                         delete_message_after=delete_message_after,
+                         disable_components_after=disable_components_after,
+                         timeout=timeout)
+        self.role = role
+        self.channel = channel
+
+        if channel:
+            self.select_callback.default_values = [channel]
+
+    @discord.ui.select(cls=discord.ui.ChannelSelect,
+                       channel_types=[discord.ChannelType.text, discord.ChannelType.private],
+                       min_values=1, max_values=1)
+    async def select_callback(self, itx: discord.Interaction[LightningBot], select: discord.ui.ChannelSelect):
+        channel = select.values[0].resolve()
+        await itx.response.defer()
+        if not channel:
+            await itx.followup.send("Unable to set the verification channel. Please try again!", ephemeral=True)
+            return
+
+        query = """INSERT INTO guild_gatekeeper_config (guild_id, verification_channel_id)
+                   VALUES ($1, $2)
+                   ON CONFLICT (guild_id)
+                   DO UPDATE SET verification_channel_id=EXCLUDED.verification_channel_id;"""
+        await itx.client.pool.execute(query, itx.guild_id, channel.id)
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (edge_case_not_handled):** Validate channel selection before proceeding.

Before executing the database query, ensure that the selected channel is valid and accessible to avoid potential issues with invalid channel IDs.

```suggestion
channel = discord.utils.get(itx.guild.text_channels, id=int(select.values[0]))
if not channel:
    await itx.followup.send("The selected channel is invalid or not accessible.", ephemeral=True)
    return
```